### ISSUE TITLE
MultitaskRegressor uncertainty prediction fails

### ISSUE THREAD

**Issue by das046** [deepchem/deepchem#2422]
## ❓ Questions & Help

Hi, 

I was running the **Uncertainty in Deep Learning** tutorial at https://github.com/deepchem/deepchem/blob/master/examples/tutorials/25_Uncertainty_In_Deep_Learning.ipynb
The code worked, but the prediction results were not good. I just followed the tutorial and then plotted the predictions and true values. See the following figure, it seems all predictions were kind of constant. 

Could you please help me with this?

Thanks.

![image](https://user-images.githubusercontent.com/50300211/109912066-39998880-7c60-11eb-9fe5-960dfb7a7d37.png)
![image](https://user-images.githubusercontent.com/50300211/109912437-eb38b980-7c60-11eb-8772-fa68d69191ae.png)

**Comment by rbharath** [comment]
Hmm, not sure why the tutorial code isn't training well. As a couple of general debugging thoughts, perhaps try lowering the learning rate? If you pass in a parameter `all_losses` to `fit()` you can get back the full loss curve and plot it to see if it's learning with something like:
```
all_losses = []
model.fit(...,all_losses=all_losses)
plt.plot(range(nb_epoch), all_losses)
```

**Comment by das046** [comment]
Thanks for the suggestions. 

I tried to lower the learning rate to 0.00001, but it didn't help. Here is the loss curve with 300 epochs and prediction over true values. 
![image](https://user-images.githubusercontent.com/50300211/110058169-bc7d1a80-7d16-11eb-9fd9-9bf9cc8863a3.png)
![image](https://user-images.githubusercontent.com/50300211/110058267-e6ced800-7d16-11eb-8f8c-56d0f367f7ed.png)
I also tried to train 3000 epochs. Here is the loss curve.
![image](https://user-images.githubusercontent.com/50300211/110058287-f1896d00-7d16-11eb-9af1-5704df4f3d86.png)
![image](https://user-images.githubusercontent.com/50300211/110058305-f77f4e00-7d16-11eb-9e6c-8d7f8d8c7eba.png)

**Comment by rbharath** [comment]
Hmm, I'm not sure what's happening here. I can try to give this a run later but I'm behind on other work this week and it might take me a bit.

CC @peastman who might have more insight into the uncertainty handling

**Comment by peastman** [comment]
I can confirm that the quality of the predictions goes down when you enable uncertainty.  I've been investigating it, but I haven't yet figured out why.

**Comment by mcloughlin2** [comment]
The OP didn't mention this, but the tutorial predictions are fine in DC version 2.1. They only become nearly constant in versions 2.3 and 2.4 (we didn't try 2.2).

**Comment by peastman** [comment]
I finally tracked this down!  The fix is in #2538.
